src/fromager/commands/build.py

```diff
@@ -165,13 +175,25 @@ def _build(
 
 def _is_wheel_built(
     wkctx: context.WorkContext, dist_name: str, dist_version: Version
-) -> bool:
+) -> tuple[bool, int]:
     req = Requirement(f"{dist_name}=={dist_version}")
 
     try:
         logger.info(f"{req.name}: checking if {req} was already built")
-        sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
-        return True
+        url, _ = sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
+        build_tag = wkctx.settings.build_tag(req.name, dist_version)
+        wheel_filename = urlparse(url).path.rpartition("/")[-1]
+        _, _, build_tag_from_name, _ = parse_wheel_filename(wheel_filename)
+        if build_tag_from_name:
+            increment_by = max(build_tag_from_name[0] - build_tag, 0)
```

Comment:
The issue was that `build_wheel` was getting the build tag from `settings.build_tag`. I am passing the `increment_by` as an amount that needs to be added to `settings.build_tag`. I guess I should maybe change the `build_wheel` command to take a `build_tag` as an argument instead of getting it directly from the settings

Reply:
now that i think of it all the time the `build_tag` should only come from settings. Only in this edge case we need a different `build_tag`. I think it might make more sense to add an optional `override_build_tag` parameter to `build_wheel` instead so that i don't have to change a lot of code